django/db/models/indexes.py

```diff
@@ -52,11 +59,23 @@ def check_name(self):
         return errors
 
     def create_sql(self, model, schema_editor, using=''):
+        def get_condition_sql():
+            # Only the WhereNode is of interest for the partial index
+            query = Query(model=model)
+            query.add_q(self.conditions)
+            compiler = query.get_compiler(connection=schema_editor.connection)
+            return query.where.as_sql(compiler=compiler, connection=schema_editor.connection)
+
         fields = [model._meta.get_field(field_name) for field_name, _ in self.fields_orders]
         col_suffixes = [order[1] for order in self.fields_orders]
+        cond_sql, cond_params = get_condition_sql()
```

Comment:
I think you could inline this function call on make it an instance method that expects `model` and `schema_editor` to be passed along.

Reply:
Seems like a fine suggestion, I just wanted to have it in a separate block.